ui/src/app/cohort-search/cohort-search/cohort-search.component.ts

```diff
@@ -42,17 +46,26 @@ export class CohortSearchComponent implements OnInit, OnDestroy {
   constructor(
     private actions: CohortSearchActions,
     private route: ActivatedRoute,
+    private workspaceStorageService: WorkspaceStorageService,
   ) {}
 
   ngOnInit() {
     console.log(`Entering CohortSearchComponent.ngOnInit with route:`);
     console.dir(this.route);
 
-    const {queryParams: query$, data: data$} = this.route;
-    this.subscription = Observable.combineLatest(query$, data$).subscribe(([params, data]) => {
+    const {queryParams: query$} = this.route;
+
+    this.workspaceStorageService.activeWorkspace$.subscribe((workspace) => {
+      this.actions.cdrVersionId = parseInt(workspace.cdrVersionId, 10);
+    });
+    this.workspaceStorageService.reloadIfNew(
+      this.route.snapshot.params['ns'],
+      this.route.snapshot.params['wsid']);
+    this.updateWrapperDimensions();
+
+    this.subscription = Observable.combineLatest(query$).subscribe(([params]) => {
```

Comment:
This change is something that John and I have talked about here: https://github.com/all-of-us/workbench/pull/794 (see the outdated comments around this same file). I'm pretty sure this change is sufficient, but the relevant thought is:

> Anyway the important thing is to make sure that this chunk of code is reliably run on route change, and I'd rather it be explicitly tied to the route than rely on the router never reusing this component, which seems less obviously certain. So lets use the observables here; whether we get them from the grandparent or by walking the pathFromRoot is less of a concern to me.